vcd/resource_vcd_org_vdc.go

```diff
@@ -536,6 +600,152 @@ func resourceVcdVdcDelete(d *schema.ResourceData, meta interface{}) error {
 	return nil
 }
 
+// updateAssignedVmSizingPolicies handles VM sizing policies.
+func updateAssignedVmSizingPolicies(vcdClient *VCDClient, d *schema.ResourceData, meta interface{}) error {
+	if vcdClient.Client.APIVCDMaxVersionIs(">= 33.0") {
+		log.Printf("[TRACE] updating assigned VM sizing policies to VDC")
+
+		_, idsOk := d.GetOk("vm_sizing_policy_ids")
+		_, defaultIdOk := d.GetOk("default_vm_sizing_policy_id")
+
+		if idsOk != defaultIdOk {
+			return fmt.Errorf("both fields are required `vm_sizing_policy_ids` and `default_vm_sizing_policy_id`")
+		}
+
+		vcdClient := meta.(*VCDClient)
+		vcdComputePolicyHref, err := vcdClient.Client.OpenApiBuildEndpoint(types.OpenApiPathVersion1_0_0, types.OpenApiEndpointVdcComputePolicies)
+		if err != nil {
+			return fmt.Errorf("error constructing HREF for compute policy")
+		}
+
+		adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingOrg, err)
+		}
+
+		vdc, err := adminOrg.GetAdminVDCByName(d.Get("name").(string), false)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingVdcFromOrg, d.Get("org").(string), d.Get("name").(string), err)
+		}
+
+		if d.HasChange("default_vm_sizing_policy_id") && !d.HasChange("vm_sizing_policy_ids") {
```

Comment:
Shoud we return early without trigerring `changeVmSizingPoliciesAndDefaultId`?

Reply:
add early return